src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -197,21 +200,24 @@ private static StudyInfos toInfos(StudyEntity entity) {
                 .build();
     }
 
-    private static BasicStudyInfos toBasicInfos(BasicStudyEntity entity) {
+    private static BasicStudyInfos toBasicInfos(StudyCreationRequestEntity entity) {
         return BasicStudyInfos.builder().studyName(entity.getStudyName())
-                .creationDate(ZonedDateTime.ofInstant(entity.getDate().toInstant(ZoneOffset.UTC), ZoneId.of("UTC")))
+                .creationDate(entity.getDate())
                 .userId(entity.getUserId())
+                .id(entity.getId())
                 .build();
     }
 
     Flux<StudyInfos> getStudyList(String userId) {
-        return studyRepository.getStudies(userId).map(StudyService::toInfos)
-                .sort(Comparator.comparing(StudyInfos::getCreationDate).reversed());
+        Stream<StudyEntity> allUserPrivateStudies = studyRepository.findAllByUserIdAndIsPrivate(userId, true).stream();
+        Stream<StudyEntity> allPublicStudies = studyRepository.findAllByIsPrivate(false).stream();
```

Comment:
is it worth it to do only one query and sort in the database ?

Reply:
it's ok